### PR TITLE
Add blog for image gallery

### PR DESCRIPTION
**Pull request by joan-ing**
## Summary

-

### Blog Release Checklist

- \[ ] Release on: \<mm/dd/yy>
- \[ ] Published on dev.to (connected to Xata org)
- \[ ] OG image included
- \[ ] Social media is scheduled for this post

### CODE REVIEW

File: build-image-gallery-astro-cloudflare.mdx [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+    Create Profile &rarr;
+  </button>
+</form>
+```
+
+## Handling image uploads server-side with the Xata SDK
+
+![(/photograph/create - where users upload their photographs)](/images/build-image-gallery/image13.png)
+
+(/photograph/create - where users upload their photographs)
```
Comment (kostasb) [inline build-image-gallery-astro-cloudflare.mdx:213]:
```suggestion
```

Comment (kostasb) [reply build-image-gallery-astro-cloudflare.mdx:213]: This is duplicate and should be removed

File: build-image-gallery-astro-cloudflare.mdx [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+        ),
+      )
+    }
+  </div>
+</Layout>
+```
+
+![(Image gallery - presenting blur images initially)](/images/build-image-gallery/image15.png)
+
+(Image gallery - presenting blur images initially)
```
Comment (kostasb) [inline build-image-gallery-astro-cloudflare.mdx:416]:
```suggestion
```

File: build-image-gallery-astro-cloudflare.mdx [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+
+![(Image gallery - after image loads complete)](/images/build-image-gallery/image15.png)
+
+(Image gallery - after image loads completely)
```
Comment (kostasb) [inline build-image-gallery-astro-cloudflare.mdx:420]:
```suggestion
```

File: build-image-gallery-astro-cloudflare.mdx [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+    }
+  </div>
+</Layout>
+```
+
+![(Image gallery - presenting blur images initially)](/images/build-image-gallery/image15.png)
+
+(Image gallery - presenting blur images initially)
+
+![(Image gallery - after image loads complete)](/images/build-image-gallery/image15.png)
```
Comment (kostasb) [inline build-image-gallery-astro-cloudflare.mdx:418]: This is the same image as the one on line 414

So:
Image gallery - presenting blur images initially
Image gallery - after image loads complete

are the same. What did we mean to show here?

File: build-image-gallery-astro-cloudflare.mdx [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+| Astro with Xata             | https://xata.io/docs/getting-started/astro                                     |
+| Astro View Transition Forms | https://docs.astro.build/en/guides/view-transitions/#transitions-with-forms    |
+| Xata File Attachments       | https://xata.io/docs/sdk/file-attachments#upload-a-file-using-file-apis        |
+| Xata Transformations        | https://xata.io/docs/sdk/image-transformations                                 |
+| Xata Get Records            | https://xata.io/docs/sdk/get#the-typescript-sdk-functions-for-querying         |
+| Cloudflare Workers Limits   | https://developers.cloudflare.com/workers/platform/limits/#request-limits      |
+
+## What's next?
+
+We'd love to hear from you if you have any feedback on this tutorial, would like to know more about Xata, or if you'd like to contribute a community blog or tutorial. Reach out to us on [Discord](https://discord.com/invite/kvAcQKh7vm) or join us on [X | Twitter](https://twitter.com/xata). Happy building 🦋
```
Comment (kostasb) [inline build-image-gallery-astro-cloudflare.mdx:456]:
```suggestion
We'd love to hear from you if you have any feedback on this tutorial, would like to know more about Xata, or if you'd like to contribute a community blog or tutorial. Reach out to us on [Discord](https://xata.io/discord) or join us on [X | Twitter](https://twitter.com/xata). Happy building 🦋
```

**Review by kostasb — CHANGES_REQUESTED** [review]
Left some inline comments most importantly: a couple images seem to be incorrect and some duplicate headers need removing.

File: build-image-gallery-astro-cloudflare.mdx [hunk 6]
```diff
@@ -0,0 +1,456 @@
...
+<html>
+  <head>
+    <ViewTransitions />
+    <!-- stuff here -->
+  </head>
+  <body>
+    <!-- stuff here -->
+  </body>
+</html>
+```
```
Comment (snide) [inline build-image-gallery-astro-cloudflare.mdx:156]: All of these references are for `astro` syntax, not `tsx` syntax.

**Review by kostasb — APPROVED** [review]
LGTM

File: build-image-gallery-astro-cloudflare.mdx [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+
+## Handling image uploads server-side with the Xata SDK
+
+![(/photograph/create - where users upload their photographs)](/images/build-image-gallery/image13.png)
+
+As Cloudflare Pages request body size allows up to 100 MB, you’ll be able to handle image uploads on the server-side. Create an Astro endpoint (`src/pages/api/upload/index.ts`) to receive POST requests containing image binaries and use the Xata SDK to store them in the `uploads` table.
+
+After doing sanity checks on the request body, first create a new (empty) record in your `uploads` table, and then use it as a reference to place the image (buffer) using the Xata TypeScript SDK. Once successfully completed, the endpoint responds with the image’s `public URL`, `height` and `width` back to the front-end to include in the form fields.
+
+```tsx
```
[the remark this answers is not in the document]
Comment (joan-ing) [reply build-image-gallery-astro-cloudflare.mdx:217]: I think this one is going in the `index.ts` file, so I thought it was Typescript. Should I change it to Astro?

**Pull request merged**